modules/api/src/main/java/org/eclipse/xpanse/api/controllers/WorkFlowApi.java

```diff
@@ -0,0 +1,97 @@
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST API methods for workflow.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse")
+@CrossOrigin
+@Secured({ROLE_ADMIN, ROLE_USER})
+public class WorkFlowApi {
+
+    @Resource
+    private WorkflowProcessUtils workflowProcessUtils;
+
+    @Resource
+    private IdentityProviderManager identityProviderManager;
+
+    /**
+     * Query process status.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Query process status by processInstanceId")
+    @GetMapping(value = "/workflow/state/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public Response queryWorkFlowById(@PathVariable("id") String processInstanceId) {
+        String workFlowState = workflowProcessUtils.getWorkFlowState(processInstanceId);
+        return Response.successResponse(List.of(workFlowState));
+    }
+
+    /**
+     * Query the tasks that need to be done by me.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Query the tasks that need to be done by me")
+    @GetMapping(value = "/workflow/task/todo", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public List<WorkFlowTask> queryTodoTasks() {
+        String userId = identityProviderManager.getCurrentUserInfo().getUserId();
+        return workflowProcessUtils.todoTasks(userId);
+    }
+
+    /**
+     * Query the tasks I have completed.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Query the tasks I have completed")
+    @GetMapping(value = "/workflow/task/done", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public List<WorkFlowTask> queryDoneTasks() {
+        String userId = identityProviderManager.getCurrentUserInfo().getUserId();
+        return workflowProcessUtils.doneTasks(userId);
+    }
+
+    /**
+     * Complete tasks based on task ID and set global process variables.
+     */
+    @Tag(name = "WorkFlow", description = "APIs to manage the WorkFlow")
+    @Operation(description = "Complete tasks by task ID and set global process variables .")
+    @PutMapping(value = "/workflow/task/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public void completeTask(@PathVariable("id") String taskId, @RequestBody Map<String,
```

Comment:
@WangLiNaruto here having a generic map will make the method generic but will be confusing. 
It would be better to have something to have like this
1. manageFailedOrder -> which should have be a wrapper to this completeTask method.
2. the manageFailedOrder will have a flag -> retryOrder/closeOrder. Based on this value, we will call the completeTask method accordingly. 

Would this be possible?